.github/workflows/snap.yml

```diff
@@ -27,19 +34,67 @@ jobs:
   test:
     runs-on: ubuntu-latest
     needs: [build]
+    outputs:
+      pebble-version: ${{ steps.install-pebble.outputs.version }}
 
     steps:
       - uses: actions/download-artifact@v3
         with:
           name: ${{ needs.build.outputs.pebble-snap }}
 
       - name: Install the Pebble snap
+        id: install-pebble
         run: |
+          set -ex
           # Install the Pebble snap from the artifact built in the previous job
           sudo snap install --dangerous --classic ${{ needs.build.outputs.pebble-snap }}
 
           # Make sure Pebble is installed
-          pebble version
+          echo "version=$(pebble version --client)" >> "$GITHUB_OUTPUT"
 
       - name: Run smoke test
-        run: pebble enter exec echo Hello | grep Hello
+        run: pebble enter --create-dirs exec echo Hello | grep Hello
+
+  promote:
+    if: ${{ github.event_name == 'release' }}
+    runs-on: ubuntu-latest
+    needs: [test]
+    strategy:
+      fail-fast: false
+      matrix:
+        arch: [amd64, arm64, ppc64el, armhf, s390x]
+    env:
+      TRACK: latest
+      DEFAULT_RISK: edge
+      TO_RISK: candidate
+    steps:
+      - name: Install Snapcraft
+        run: sudo snap install snapcraft --classic
+      
+      - name: Wait for ${{ needs.test.outputs.pebble-version }} to be released
+        env:
+          SNAPCRAFT_STORE_CREDENTIALS: ${{ secrets.SNAPCRAFT_STORE_CREDENTIALS }}
+        run: |
+          while ! `snapcraft status ${{ env.SNAP_NAME }} --track ${{ env.TRACK }} --arch ${{ matrix.arch }} \
+              | grep "${{ env.DEFAULT_RISK }}" \
+              | awk -F' ' '{print $2}' \
```

Comment:
I'm curious why `-F' '` is necessary when the default for AWK is to split on (run of) whitespace.

Reply:
it's not :) It's just a force of habit and a preference to be explicit.